### PR TITLE
Do not throw error on duplicate service registration.

### PR DESCRIPTION
**Pull request by hsubox76**
A change needed for when multiple components each internally depend on another component and may each try to register it.

### CODE REVIEW

File: packages/app/src/firebaseNamespaceCore.ts [hunk 1]
```diff
@@ -182,7 +182,10 @@ export function createFirebaseNamespaceCore(
   ): FirebaseServiceNamespace<FirebaseService> {
     // Cannot re-register a service that already exists
```
Comment (Feiyang1) [inline packages/app/src/firebaseNamespaceCore.ts]: The comment is outdated and needs update.

Comment (Feiyang1) [reply packages/app/src/firebaseNamespaceCore.ts]: I think we should log a message at DEBUG log level for catching the possible version mixup. We already use logger in `@firebase/app`, we can make it sharable and use it here.

File: packages/app/test/firebaseApp.test.ts [hunk 2]
```diff
@@ -87,6 +87,32 @@ function executeFirebaseTests(): void {
...
+
+      (firebase as _FirebaseNamespace).INTERNAL.registerService(
+        'test',
+        (app: FirebaseApp) => {
+          registrations += 1;
+          return new TestService(app);
+        }
+      );
+
+      (firebase as any).test();
```
Comment (Feiyang1) [inline packages/app/test/firebaseApp.test.ts]: We should just test no exception is thrown the second time and that `firebase.test === the previous firebase.test`

File: packages/app/src/firebaseNamespaceCore.ts (hunk 1 above)
Comment (hsubox76) [reply packages/app/src/firebaseNamespaceCore.ts]: Done.

File: packages/app/test/firebaseApp.test.ts (hunk 2 above)
Comment (hsubox76) [reply packages/app/test/firebaseApp.test.ts]: Added equality test.  To check for no exception, do we have to do anything?  Or just leave it alone and throwing will fail the test?

Comment (hsubox76) [reply packages/app/test/firebaseApp.test.ts]: Updated test to check for exception and service namespace equality.

**Comment by lookfirst** [comment]
THANK YOU!

**Pull request merged**